src/updater.tsx

```diff
@@ -1,14 +1,21 @@
-import React, { useEffect, useMemo, useRef } from 'react'
```

Comment:
No functional changes in this file, just reorganized the big useEffect hook code for legibility